pkg/providers/amifamily/ami.go

```diff
@@ -233,14 +276,18 @@ func getFiltersAndOwners(amiSelector map[string]string) ([]*ec2.Filter, []*strin
 	return filters, owners
 }
 
-func sortAMIsByCreationDate(amiRequirements map[AMI]scheduling.Requirements) []AMI {
-	amis := lo.Keys(amiRequirements)
-
+func sortAMIsByCreationDate(amis []AMI) []AMI {
 	sort.Slice(amis, func(i, j int) bool {
-		itime, _ := time.Parse(time.RFC3339, amis[i].CreationDate)
-		jtime, _ := time.Parse(time.RFC3339, amis[j].CreationDate)
-		return itime.Unix() >= jtime.Unix()
+		if amis[i].CreationDate != "" {
+			itime, _ := time.Parse(time.RFC3339, amis[i].CreationDate)
+			jtime, _ := time.Parse(time.RFC3339, amis[j].CreationDate)
```

Comment:
If users are able to hack in the CreationDate to be undefined, does that mean a user can potentially hack in a different formatted CreationDate too? 

I haven't ever seen a nil CreationDate on an image, but I do see that it isn't required. I would be interested to know if we can definitively say that we'll never get an incorrectly formatted CreationDate, and if not, why we shouldn't be checking the error here.

Reply:
If a creationDate is not able to be parsed, then the unix time will be zero and the ami will be sorted to being in the bottom of the list.